### ISSUE TITLE
Use CLDR data for translations instead of manual 

### ISSUE THREAD

**Issue by damienalexandre** [maoschanz/emoji-selector-for-gnome#61]
I see that search is based on some translations contributed here, this can lead to various problems:

- you can only support a limited number of languages;
- not all emoji have their translations;
- the translation / annotation of an emoji can differ from what users expect / have seen elsewhere.

But the thing is, Unicode and CLDR provides annotations for emoji in A LOT of languages, and for ALL emoji, in a easy to process format: https://www.unicode.org/repos/cldr/tags/release-34/common/annotations/en.xml

So I think this extension should rely on the official emoji annotation instead of trying to translate all emojis.

I do a bit of processing on the CLDR XML files here: https://github.com/jolicode/emoji-search - you can find Lucene formatted synonym files looking like this:

```
👩‍🚒 => 👩‍🚒, firefighter, firetruck, woman
👩‍✈ => 👩‍✈, pilot, plane, woman
🥓 => 🥓, bacon, meat, food
🥔 => 🥔, potato, vegetable, food
😅 => 😅, cold, face, open, smile, sweat
😆 => 😆, face, laugh, mouth, open, satisfied, smile
🚎 => 🚎, bus, tram, trolley
🇫🇷 => 🇫🇷, france
🇬🇧 => 🇬🇧, united kingdom
```

Or just replace /update-and-compile-translations.sh with a new logic.

Cheers.

**Comment by maoschanz** [comment]
>search is based on some translations contributed here

bold of you to assume these contributions exist, iirc there is only the french one from me, and maybe italian but i haven't taken the time to merge it yet, but they are mainly copy-paste of these CLDR data.

There are 2 issues here:

- i use gettext to translate because i've no idea how to load things otherwise (i guess i should manually get environment variables etc.)
- the current system for skin tones, gender etc. is using custom keywords, while CLDR has one line per variant, if i change this, everything from displaying rows to copying emojis has to be different

Using these data would need an rewrite of half of the code, it's unlikely that i take time to do it while i'm still playing with my very cool Paint clone.

>Or just replace /update-and-compile-translations.sh with a new logic.

Not "replace", this extension will still needs string other than keywords for searching emojis

**Comment by amivaleo** [comment]
[ot]

> Using these data would need an rewrite of half of the code, it's unlikely that i take time to do it while i'm still playing with my very cool Paint clone.

Sorry for bothering here. Github has no private messaging so I didn't know how/where to say this: that project is AMAZING.
Thank you SO MUCH for working on that. :o

[/ot]

**Comment by damienalexandre** [comment]
You know this is only a suggestion for enhancement and not a criticism or a personal attack? The tone of your response feels angry to me and I was not expecting that. Be sure I'm not here to give orders or ask for anything, I love this extension and use it daily on my computer. I know this can't be done that easily and if you don't want / don't have the time, that's no problem. Sorry for my poor choice of words that may have caused some confusion :heart: 

### To get back to the suggestion

IMO the interface translations and the emoji search mechanism could be separated, and the benefits will be a broader / more complete search experience for users.

Yes as you say this will greatly impact the extension.

If you think this is a bad idea feel free to close.

Cheers.

**Comment by maoschanz** [comment]
>You know this is only a suggestion for enhancement and not a criticism or a personal attack?

Yes i know, but i feel like i have to explain why i will not do it right now despite the fact it's a very good idea
